packages/cache/src/internal/cacheUtils.ts

```diff
@@ -102,10 +102,14 @@ export async function getCompressionMethod(): Promise<CompressionMethod> {
   const version = semver.clean(versionOutput)
   core.debug(`zstd version: ${version}`)
 
-  if (versionOutput === '') {
+  if (version === null) {
     return CompressionMethod.Gzip
-  } else {
+  } else if (semver.lt(version, '1.3.2')) {
     return CompressionMethod.ZstdWithoutLong
+  } else if (semver.lt(version, '1.3.6')) {
+    return CompressionMethod.ZstdWithoutAdapt
+  } else {
+    return CompressionMethod.Zstd
```

Comment:
This should also fix this bug, as it seems we have not actually been able to use long mode yet.